packages/muelu/utils/MergeLevelGraphs/mergeLevelGraphs.py

```diff
@@ -0,0 +1,320 @@
+
+				for edge in edgeLst:
+					# map the nodes to their new connections
+					POINTS = edge.obj_dict["points"]
+
+					if((searchPoint != "") and ((searchPoint in POINTS) == False)):
+						continue
+
+					self.__NODE_0 = self.__subG.get_node(POINTS[0])
+					self.__NODE_1 = self.__subG.get_node(POINTS[1])
+
+					if len(self.__NODE_0) < 1:
+						self.__NODE_0 = self.__findNode(POINTS[0])
+					elif len(self.__NODE_1) < 1:
+						self.__NODE_1 = self.__findNode(POINTS[1])
+
+					# connect repeat nodes to it's counter part
+					if LABEL == self.__NODE_0[0].obj_dict["attributes"]["label"]:
+						searchPoint = POINTS[0]
+						self.__replaceConnection(LABEL,
+							edge.obj_dict["attributes"])
+
+					elif LABEL == self.__NODE_1[0].obj_dict["attributes"]["label"]:
+						searchPoint = POINTS[1]
+						self.__replaceConnection(LABEL,
+							edge.obj_dict["attributes"], False)
+					
+					if searchPoint != "":
+						if (searchPoint in self.__delNodes) == False: 
+							self.__delNodes.append(searchPoint)
+
+			self.__deleteNodes()
+			# add subgraph to masterGraph
+			self.__masterGraph.add_subgraph(self.__subG)
+			self.__graphCnt += 1
+		self.__renameSubgraphs()
+	
+	def __getFileName(self, PATH: str, ext: str=".dot") -> int:
+		'''
+		Finds a file number that does not exists and sets fileCnt to it.
+		# Params:
+		PATH - file path to save the .dot/.png file\n
+		fileName - the file extension determin if an image file
+		exists or a dot file exists\n
+		# Return:
+		file number that does not exist
+		'''
+		fileCnt = 0
+
+		while os.path.exists(os.path.join(PATH, f"{fileCnt}{ext}")):
+			fileCnt += 1
+		return fileCnt
+	
+	def write(self) -> None:
+		'''Save each subgraph to a file.'''
+		PATH = os.path.dirname(__file__)
+		graphFile = ""
+		imageFile = ""
+
+		if "-o" in sys.argv:
+			try:
+				graphFile = sys.argv[-2]
+			except:
+				raise RuntimeError("Pass two file paths in the command line arguments to save to file")
+			
+			imageFile = sys.argv[-1]
+
+			if os.access(graphFile, os.X_OK) == False:
+				if graphFile[-4:] != ".dot":
+					raise RuntimeError(f"File must be a .dot file {graphFile}")
+			
+			if os.access(imageFile, os.X_OK) == False:
+				if imageFile[-4:] != ".png":
+					raise RuntimeError(f"File must be a .png file {imageFile}")
+
+			while os.path.exists(graphFile):
+				ans = input(f"Are you sure you want to overwrite this file: {graphFile}\n Enter \"y\" to pick a different file name: ")
+				if ans.strip().lower() == "y":
+					ans = input("Enter file path: ")
+					graphFile = ans
+				else:
+					break
+
+			while os.path.exists(imageFile):
+				ans = input(f"Are you sure you want to overwrite this file: {imageFile}\n Enter \"y\" to pick a different file name: ")
+				if ans.strip().lower() == "y":
+					ans = input("Enter file path: ")
+					imageFile = ans
+				else:
+					break
+		else:
+			graphFile = os.path.join(PATH, "output.dot")
+			imageFile = os.path.join(PATH, "output.png")

```

Comment:
The output is saved to the location of the script. That seems non-standard and leads to unintuitive behavior. Rather use the location from where the script is called.